### PR TITLE
🔧 switch to tree-sitter-typescript

### LINKED ISSUE #2788: task: switch to tree-sitter-typescript
https://github.com/tree-sitter/tree-sitter-typescript

this NPM package can replace the language-typescript package which blueprint currently pulls down from github

### PR DESCRIPTION
**Pull request by giladgray**
#### Fixes #2788 

also move syntax dependencies to docs-data, the only place they are used.

### CODE REVIEW

File: packages/docs-data/package.json [hunk 1]
```diff
@@ -13,10 +13,13 @@
     "@blueprintjs/core": "^3.0.0",
     "@blueprintjs/docs-theme": "^3.0.0",
     "documentalist": "^1.4.0",
+    "better-handlebars": "github:wmeldon/better-handlebars",
     "glob": "^7.1.2",
     "highlights": "^3.1.1",
+    "language-less": "github:atom/language-less",
```
Comment (adidahiya) [inline packages/docs-data/package.json]: can we find a published version of this syntax definition as well?

File: packages/docs-data/docsUtils.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const marked = require("marked");
 const Highlights = require("highlights");
 
 const DEFAULT_SCOPE = "source.tsx";
-const HIGHLIGHTS_LANGUAGES = ["better-handlebars", "language-typescript", "language-less"];
+const HIGHLIGHTS_LANGUAGES = ["better-handlebars", "language-less", "tree-sitter-typescript"];
```
Comment (adidahiya) [inline packages/docs-data/docsUtils.js]: where do we actually use handlebars in code examples?

Comment (giladgray) [reply packages/docs-data/docsUtils.js]: every html snippet, for `{{:modifier}}` support

File: packages/docs-data/package.json (hunk 1 above)
Comment (giladgray) [reply packages/docs-data/package.json]: i looked but it's only published to atom's `apm`, not npm.

**Pull request merged**